server/vcs/provider/github/pr_fetcher.go

```diff
@@ -0,0 +1,29 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	gh "github.com/google/go-github/v45/github"
+	"github.com/palantir/go-githubapp/githubapp"
+	"github.com/pkg/errors"
+	"net/http"
+)
+
+type PRStateFetcher struct {
+	ClientCreator githubapp.ClientCreator
+}
+
+func (c *PRStateFetcher) FetchLatestState(ctx context.Context, installationToken int64, repoOwner string, repoName string, prNum int) (*gh.PullRequest, error) {
```

Comment:
nit: I think an appropriate name would just be PRFetcher, and the method just be Fetch since the returned object is just a pull request and has no state in the name.